drivers.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2022 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef PV_DRIVERS_H
+#define PV_DRIVERS_H
+
+#include <stdlib.h>
+#include <limits.h>
+
+#include "pantavisor.h"
+
+typedef enum {
+	MOD_LOAD = 0,
+	MOD_UNLOAD
+} mod_action_t;
+
+typedef enum {
+	MOD_UNKNOWN = -1,
+	MOD_UNLOADED,
+	MOD_LOADED
+} mod_state_t;
+
+struct pv_driver {
+	char *alias;
+	char **modules;
+	bool loaded;
+	struct dl_list list;
+};
+
+const char *pv_drivers_state_str(char *match);
+int pv_drivers_state(char *match);
+int pv_drivers_load(char *match);
+int pv_drivers_unload(char *match);
+
+struct pv_driver* pv_drivers_add(struct pv_state *s, char *alias, int len, char **modules);
+void pv_drivers_remove(struct pv_driver *d);
```

Comment:
maybe call this one from pv_state_free

Reply:
i think this is done now too. check it.

Reply:
Looks good now. I guess we could also remove pv_drivers_remove as that's not used anywhere